server/api/impl/aws/provision_app.go

```diff
@@ -103,7 +122,20 @@ func (c *Client) ProvisionApp(releaseURL string) error {
 			}
 		}
 	}
-	return c.provisionApp(mani.AppID, mani.Version, resFunctions)
+
+	newManifest, err := c.provisionAssets(&mani, assets)
+	if err != nil {
+		return errors.Wrapf(err, "can't provision assets of the app - %s", mani.AppID)
```

Comment:
might be helpful to add an error wrapper across the PP (similar to Lev's approach in https://github.com/mattermost/mattermost-webapp/pull/7426/commits/15d9c853c6aea0b7097aa78a10a1eeb170fc7d4a
 - see `asError/errorMessage()`
 to consolidate the capture of `appId` (and potentially other metadata)

Reply:
1/5 that the caller should add metadata like this that is tied to a function call.

Reply:
If we had code with an `app` context I'd agree. But it seems like there is not much. Just an install, provision part. And they all get `app` as an argument. So `appID` is not a state.

Reply:
My point was more towards consolidating the collection of metadata for error/logging purposes. Right now, in different places we collect diff. pieces of metadata to provide the best context for an error,  like `appId`, `file name`, `key`, or none. In some cases we can/makes sense to collect one piece of metadata, in others, multiple. I just want us to make sure that we collect the best context for each and do not miss smth. that could help the debugging process later.